cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,401 @@
+
+		err = json.Unmarshal(rr.Body.Bytes(), &actualData)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if rr.Code == http.StatusOK && response.StatusCode == 0 {
+			logrus.WithFields(logrus.Fields{"calculation": actualData.Name, "teff": actualData.Spec.Teff, "logG": actualData.Spec.LogG}).Info("Created calculation using the api server...")
+		} else {
+			logrus.Info("No calculation was created...")
+		}
+	}
+}
+
+func TestDeleteCalculation(t *testing.T) {
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	teff := 10000
+	testCalculationCount := teff + 2
+	var calcList []*calculationsv1.Calculation
+	for teff != testCalculationCount {
+		teff++
+		newCalc := util.NewCalculation(float64(teff), 4.0)
+
+		calcList = append(calcList, newCalc)
+		logrus.WithFields(logrus.Fields{"calculation": newCalc.Name, "teff": teff, "logG": "4.0"}).Info("Creating calculation for test purpose")
+		_, err := fakeClient.Calculations().Create(ctx, newCalc, metav1.CreateOptions{})
+		if err != nil {
+			t.Errorf("Coudn't create the calculation: %v", newCalc)
+		} else {
+			logrus.WithFields(logrus.Fields{"calculation": newCalc.Name, "teff": newCalc.Spec.Teff, "logG": newCalc.Spec.LogG}).Info("Calculation created successfully")
+		}
+	}
+
+	s := server{
+		logger: logrus.WithField("test-name", "get calculations test"),
+		ctx:    ctx,
+		client: fakeClient,
+	}
+
+	req, err := http.NewRequest("DELETE", fmt.Sprintf("/calculations/delete/%v", calcList[0].Name), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	q := req.URL.Query()
+	q.Add("name", calcList[0].Name)
+	req.URL.RawQuery = q.Encode()
+
+	req.Header.Set("Content-Type", "application/json")
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(s.deleteCalculation)
+	handler.ServeHTTP(rr, req)
+
+	response := Response{}
+	err = json.Unmarshal(rr.Body.Bytes(), &response)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if status := response.StatusCode; status != http.StatusOK && status != 0 {
+		logrus.Infof("Handler returned wrong status code: got %v want %v", status, http.StatusOK)
+	}
+
+	var actualData *calculationsv1.CalculationList
+
+	err = json.Unmarshal(rr.Body.Bytes(), &actualData)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if diff := cmp.Diff(calcList[1], actualData.Items); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestGetCalculations(t *testing.T) {
+	testCases := []struct {
+		id               string
+		calculationCount int
+		expected         []calculationsv1.Calculation
+	}{
+		{
+			id:               "get no calculation",
+			calculationCount: 0,
+		},
+		{
+			id:               "get two calculations",
+			calculationCount: 2,
+		},
+	}
+
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	teff := 10000
+
+	for _, tc := range testCases {
```

Comment:
we don't need that. Your testcase should hold the calculation and you create them.